src/Functions/formatQuery.cpp

```diff
@@ -0,0 +1,110 @@
+#include <Columns/ColumnString.h>
+#include <Functions/FunctionFactory.h>
+#include <IO/WriteBufferFromVector.h>
+#include <Interpreters/Context.h>
+#include <Parsers/ParserQuery.h>
+#include <Parsers/formatAST.h>
+#include <Parsers/parseQuery.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+class FunctionFormatQuery : public IFunction
+{
+public:
+    static constexpr auto name = "formatQuery";
+    static FunctionPtr create(ContextPtr context)
+    {
+        const auto & settings = context->getSettings();
+        return std::make_shared<FunctionFormatQuery>(settings.max_query_size, settings.max_parser_depth);
+    }
+
+    FunctionFormatQuery(size_t max_query_size_, size_t max_parser_depth_)
+        : max_query_size(max_query_size_), max_parser_depth(max_parser_depth_)
+    {
+    }
+
+    String getName() const override { return name; }
+
+    size_t getNumberOfArguments() const override { return 1; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
```

Comment:
I would do the type check similar to this: https://github.com/ClickHouse/ClickHouse/blob/master/src/Functions/timestamp.cpp#L41-L52